### ISSUE TITLE
Crash when date format is Double

### ISSUE THREAD

**Issue by royipressburger** [cognitect-labs/aws-api#36]
Hi all.

Im working with this lib for `cognito-idp` and I have encountered an exception when parsing the aws response: 
```
java.lang.Double cannot be cast to java.lang.CharSequence
        [clojure.core$re_matcher invokeStatic "core.clj" 4849]
        [clojure.core$re_matches invokeStatic "core.clj" 4886]
        [clojure.core$re_matches invoke "core.clj" 4886]
        [cognitect.aws.shape$parse_date invokeStatic "shape.clj" 86]
        [cognitect.aws.shape$parse_date invoke "shape.clj" 79]
        [cognitect.aws.shape$eval26825$fn__26826 invoke "shape.clj" 150]
        [clojure.lang.MultiFn invoke "MultiFn.java" 234]
```

After some debugging I saw that the response of the aws request looks like this:
```
;;Requesting a list group of a user pool, This throws the error above
(aws/invoke cognito {:op :ListGroups :request {:UserPoolId "my-user-pool-id"}})

;; Added this for debugging
(def http-meta (slurp (:body (:http-response (meta (aws/invoke cognito {:op :ListGroups :request {:UserPoolId "my-user-pool-id"}}))))))

(def response (json/parse-string http-meta true))

;; the response looks like: 
{:Groups
 [{:CreationDate 1.547453349895E9,
   :Description "My App",
   :GroupName "Group 1",
   :LastModifiedDate 1.547453349895E9,
   :UserPoolId "my-user-pool-id"}]}
```

As you can see the format of `CreationDate` and `LastModifiedDate` is `double`
this causes to crash in the `shape/parse-data`.
adding a `(double?  data) (java.util.Date. (* 1000 (long data)))` is fixing the issue

if more info is needed let me know,
Thanks ahead

**Comment by dchelimsky** [comment]
Would you please post the raw string? e.g. `(slurp (:body (:http-response (meta (aws/invoke cognito {:op :ListGroups :request {:UserPoolId "my-user-pool-id"}})))))`

**Comment by dchelimsky** [comment]
Never mind. I've got an example working.

**Comment by dchelimsky** [comment]
You made it kind of easy ;)

**Comment by dchelimsky** [comment]
Released in 0.8.204.
